common/src/main/java/tachyon/client/BlockMasterClient.java

```diff
@@ -0,0 +1,193 @@
+
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.MasterClient;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.BlockInfo;
+import tachyon.thrift.BlockMasterService;
+import tachyon.thrift.Command;
+import tachyon.thrift.NetAddress;
+import tachyon.thrift.WorkerInfo;
+
+/**
+ * The BlockMaster client, for clients.
+ *
+ * Since thrift clients are not thread safe, this class is a wrapper to provide thread safety.
+ */
+// TODO: better deal with exceptions.
+public final class BlockMasterClient extends MasterClient {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private BlockMasterService.Client mClient = null;
+
+  public BlockMasterClient(InetSocketAddress masterAddress, ExecutorService executorService,
+      TachyonConf tachyonConf) {
+    super(masterAddress, executorService, tachyonConf);
+  }
+
+  @Override
+  protected String getServiceName() {
+    return Constants.BLOCK_MASTER_SERVICE_NAME;
+  }
+
+  @Override
+  protected void afterConnect() {
+    mClient = new BlockMasterService.Client(mProtocol);
+  }
+
+  @Override
+  protected void afterDisconnect() {
+  }
+
+  /**
+   * Get the info of a list of workers.
+   *
+   * @return A list of worker info returned by master
+   * @throws IOException
+   */
+  public synchronized List<WorkerInfo> getWorkerInfoList() throws IOException {
+    while (!mIsClosed) {
+      connect();
+
+      try {
+        return mClient.getWorkerInfoList();
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return null;
+  }
+
+  public synchronized BlockInfo getBlockInfo(long blockId) throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getBlockInfo(blockId);
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the total capacity in bytes.
+   *
+   * @return capacity in bytes
+   * @throws IOException
+   */
+  public synchronized long getCapacityBytes() throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getCapacityBytes();
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Get the amount of used space in bytes.
+   *
+   * @return amount of used space in bytes
+   * @throws IOException
+   */
+  public synchronized long getUsedBytes() throws IOException {
+    while (!mIsClosed) {
+      connect();
+      try {
+        return mClient.getUsedBytes();
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    return -1;
+  }
+
+  public synchronized void workerCommitBlock(long workerId, long usedBytesOnTier, int tier, long
```

Comment:
Confused. I thought this class is only used by clients. But it seems here it provides APIs for worker.